halo2_proofs/src/circuit/layouter.rs

```diff
@@ -158,13 +165,22 @@ impl From<Selector> for RegionColumn {
     }
 }
 
+impl From<DynamicTable> for RegionColumn {
+    fn from(table: DynamicTable) -> RegionColumn {
+        RegionColumn::TableTag(table)
+    }
+}
+
 impl Ord for RegionColumn {
     fn cmp(&self, other: &Self) -> cmp::Ordering {
         match (self, other) {
             (Self::Column(ref a), Self::Column(ref b)) => a.cmp(b),
             (Self::Selector(ref a), Self::Selector(ref b)) => a.0.cmp(&b.0),
-            (Self::Column(_), Self::Selector(_)) => cmp::Ordering::Less,
+            (Self::TableTag(ref a), Self::TableTag(ref b)) => a.cmp(b),
+            (Self::Column(_), _) => cmp::Ordering::Less,
             (Self::Selector(_), Self::Column(_)) => cmp::Ordering::Greater,
+            (Self::TableTag(_), _) => cmp::Ordering::Greater,
+            (_, Self::TableTag(_)) => cmp::Ordering::Less,
```

Comment:
```suggestion
            #[cfg(feature = "unstable-dynamic-lookups")]
            (Self::TableTag(_), _) => cmp::Ordering::Greater,
            #[cfg(feature = "unstable-dynamic-lookups")]
            (_, Self::TableTag(_)) => cmp::Ordering::Less,
```
(or whatever is necessary to toggle between sorting with and without table tags).